src/main/java/com/teragrep/pth10/steps/tokenizer/RegexTokenizerUDF.java

```diff
@@ -0,0 +1,95 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth10.steps.tokenizer;
+
+import com.typesafe.config.Config;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.expressions.UserDefinedFunction;
+import org.apache.spark.sql.functions;
+import org.apache.spark.sql.types.DataTypes;
+
+public class RegexTokenizerUDF implements TokenizerApplicable {
+
+    private final String BLOOM_PATTERN_CONFIG_ITEM = "dpl.pth_06.bloom.pattern";
+    private final Config config;
+    private final AbstractTokenizerStep.TokenizerFormat format;
+    private final String inputCol;
+    private final String outputCol;
+
+    public RegexTokenizerUDF(
+            Config config,
+            AbstractTokenizerStep.TokenizerFormat format,
+            String inputCol,
+            String outputCol
+    ) {
+        this.config = config;
+        this.format = format;
+        this.inputCol = inputCol;
+        this.outputCol = outputCol;
+    }
+
+    public Dataset<Row> appliedDataset(final Dataset<Row> dataset) {
+        final String pattern = config.getString(BLOOM_PATTERN_CONFIG_ITEM).trim();
+        final UserDefinedFunction regexUDF = functions
+                .udf(
+                        new com.teragrep.functions.dpf_03.RegexTokenizerUDF(),
+                        DataTypes.createArrayType(DataTypes.BinaryType, false)
+                );
+
+        if (this.format == AbstractTokenizerStep.TokenizerFormat.BYTES) {
+            return dataset
+                    .withColumn("regex_column", functions.lit(pattern))
+                    .withColumn(outputCol, regexUDF.apply(functions.col(inputCol), functions.col("regex_column")))
```

Comment:
I think it should be possible to use `functions.lit(pattern)` as the second parameter of regexUDF, could avoid creating and dropping `regex_column`

Reply:
correct, implemented literal column straight to function